pkg/secrets/gcp.go

```diff
@@ -1,8 +1,58 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package secrets
 
+import (
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+
+	reposerver "github.com/kanisterio/kanister/pkg/secrets/repositoryserver"
+)
+
 const (
 	// GCPProjectID is the config map key for gcp project id data
 	GCPProjectID string = "gcp_project_id"
 	// GCPServiceKey is the config map key for gcp service key data
 	GCPServiceKey string = "gcp_service_key"
+	// GCPServerAccountJsonKey is the key for gcp service account json
+	GCPServiceAccountJsonKey string = "service-account.json"
+
+	// GCPSecretType represents the secret type for GCP credentials.
+	GCPSecretType string = "secrets.kanister.io/gcp"
 )
+
+// ValidateGCPCredentials function is to verify the schema of GCP secrets
+// that need to be provided for kopia commands
+// Required fields:
+// - GCPProjectID
+// - GCPServiceAccountJsonKey
```

Comment:
Should we be more descriptive about this? At first glance, it looks like we are defining the function arguments.
Could be moved inside the function too